### PR TITLE
Remove @fluidframework/view-interfaces package

### PR DESCRIPTION
**Pull request by ChumpChief**
After moving the view-adapters content to example-utils and demoting webpack-fluid-loader example, we can now move the view-interfaces content to example-utils to match the adapters and eliminate the view-interfaces package altogether.

The IFluidMountableViewClass wasn't needed anymore and is removed (it was originally used when we needed to support the request pattern, to aid in generating the appropriate request handler).  Also removed the IFluidMountableView const which seems unnecessary at this point.

Also removed a stale doc page referring to old interfaces that already don't exist anymore.

[AB#7237](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7237)

### CODE REVIEW

**Comment by RishhiB** [comment]
Should the view-interfaces package be deprecated before being removed? Should this be added to BREAKING.md?

**Comment by Josmithr** [comment]
> Did we change the package removal process? I remember we used to push only breaking changes to the next/lts branch. Do breaking changes get pushed to main now? Should the view-interfaces package be deprecated before being removed? Should this removal be added to BREAKING.md?

The process has changed, though it looks like [our docs](https://github.com/microsoft/FluidFramework/wiki/Breaking-vs-Non-breaking-Changes) need to be updated to reflect this. The `next` branch is no longer exists, except for historical context. Until we release 2.0, breaking changes may go into main. After 2.0, breaking changes to `@public` APIs will be strictly not allowed until we start prepping a 3.0 branch (probably a year or so later).

File: examples/utils/example-utils/api-report/example-utils.api.md [hunk 1]
```diff
@@ -46,6 +45,15 @@ export interface IDetachedModel<ModelType> {
     model: ModelType;
 }
 
+// @internal (undocumented)
```
Comment (Josmithr) [inline examples/utils/example-utils/api-report/example-utils.api.md:48]: I know these are all internal, but can we document them while we're here?

**Review by Josmithr — APPROVED** [review]
Left a documentation suggestion, but otherwise looks good!

File: examples/utils/example-utils/api-report/example-utils.api.md (hunk 1 above)
Comment (ChumpChief) [reply examples/utils/example-utils/api-report/example-utils.api.md]: Documented in latest - also removed the const entirely.

**Pull request merged**